prereise/gather/griddata/hifld/data_process/profiles.py

```diff
@@ -100,3 +104,119 @@ def build_wind(wind_plants, download_directory, year):
+    """Given hydro generation profiles (MW) for a set of balancing authorities, create
+    profiles which are capacity-normalized for a superset of balancing authorities.
+    Profiles are normalized either using the total capacity of the balancing authority's
+    hydro capacity or by peak generation, whichever is greater. Balancing authorities
+    which are missing from the source generation data are populated using a profile
+    aggregated from all available balancing authorities.
+
+    :param pandas.DataFrame hydro_plants: data frame of hydro plant data, including
+        columns 'Balancing Authority Code' and 'Pmax'.
+    :param pandas.DataFrame generation: data frame of hourly generation. Index is
+        timestamps, columns are balancing authority codes, values are total generation.
+    :param set query_regions: superset of balacing authority areas to produce profiles
+        for.
+    :param int/str year: year to use for output profile timestamps.
+    :return: (*pandas.DataFrame*) -- index is hourly timestamps for the given year,
+        columns are the superset balancing authorities, values are normalized
+        generation.
+    """
+    # Filter generation anomalies based on installed capacity
+    ba_groups = hydro_plants.groupby("Balancing Authority Code")
+    ba_capacities = ba_groups["Pmax"].sum()
+
+    # Normalize regional totals by regional capacities (or peak generation)
+    normalized_by_generation = generation.columns[
+        generation.max() > ba_capacities.loc[generation.columns]
+    ]
+    if len(normalized_by_generation) > 0:
+        print("regions normalized by generation profile:")
+        print(
+            generation[normalized_by_generation].max()
+            / ba_capacities.loc[normalized_by_generation]
+        )
+    normalizing_capacity = pd.concat(
+        [ba_capacities.loc[generation.columns], generation.max()],
+        axis=1,
+    ).max(axis=1)
+    normalized_profiles = generation / normalizing_capacity
+
+    # Build shape for 'default' (weighted average of other shapes)
+    normalized_profiles["default"] = (
+        normalized_profiles * ba_capacities.loc[normalized_profiles.columns]
+    ).sum(axis=1) / ba_capacities.loc[normalized_profiles.columns].sum()
+
+    # Build the output dataframe, and populate it
+    profiles = pd.DataFrame(
+        index=pd.date_range(start=f"{year}-01-01-00", end=f"{year}-12-31-23", freq="H"),
+        columns=hydro_plants.index,
+    )
+    for region in query_regions:
+        plant_ids = ba_groups.get_group(region).index
+        if region in normalized_profiles:
+            template = normalized_profiles[region]
+        else:
+            print(region, "uses 'default' hydro profile")
+            template = normalized_profiles["default"]
+        # pandas won't broadcast for us, so we need to construct a 2D object
+        profiles.loc[:, plant_ids] = pd.concat(
+            [template] * len(plant_ids), axis=1
+        ).values
+
```

Comment:
Looks we are missing the return of the profiles.

Reply:
Fixed.